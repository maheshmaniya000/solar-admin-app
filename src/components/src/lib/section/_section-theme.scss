@use '~@angular/material' as mat;

@mixin theme($theme-or-color-config) {
  $color-config: mat.get-color-config($theme-or-color-config);
  $typography-config: mat.get-typography-config($theme-or-color-config);
  $background: map-get($color-config, background);
  $foreground: map-get($color-config, foreground);

  sg-section {
    background-color: mat.get-color-from-palette($background, card);

    .header {
      border-bottom: 1px solid mat.get-color-from-palette($foreground, divider);
    }
  }
}
