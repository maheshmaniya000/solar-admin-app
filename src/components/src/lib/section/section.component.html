<h6 class="header" *ngIf="headerText">{{ headerText }}</h6>
<section>
  <ng-content></ng-content>
</section>
