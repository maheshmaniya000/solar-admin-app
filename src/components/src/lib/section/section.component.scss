:host {
  display: contents;
}

section {
  margin-bottom: 24px;
}

.header {
  display: flex;
  align-items: center;
  height: 36px;
}
