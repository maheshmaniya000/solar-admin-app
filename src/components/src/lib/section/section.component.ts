import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'sg-section',
  templateUrl: './section.component.html',
  styleUrls: ['./section.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class SectionComponent {
  @Input() headerText: string;
}
