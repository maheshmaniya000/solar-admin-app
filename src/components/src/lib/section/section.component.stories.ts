import { CommonModule } from '@angular/common';
import { MatInputModule } from '@angular/material/input';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { moduleMetadata, Story } from '@storybook/angular';

import { SgDefaultsModule } from '../defaults/defaults.module';
import { FormFieldModule } from '../form-field/form-field.module';
import { SectionModule } from './section.module';

export default {
  title: 'Solargis/Layout/Section',
  decorators: [
    moduleMetadata({
      imports: [BrowserAnimationsModule, MatInputModule, FormFieldModule, SgDefaultsModule, CommonModule, SectionModule]
    })
  ]
};

const headerText = {
  name: 'Header text',
  control: { type: 'text' }
};

export const basic: Story = props => ({
  props,
  template: `
    <div class="sg-container">
      <sg-section [headerText]="headerText">
        <div class="sg-grid-8-15">
          <div class="row-15">
            <label for="input-1">Label: </label>
            <mat-form-field>
              <input id="input-1" matInput placeholder="placeholder">
            </mat-form-field>
          </div>
          <mat-divider class="row-full-width" *ngIf="dividerVisible"></mat-divider>
          <div class="row-15">
            <label for="input-2">Label: </label>
            <mat-form-field>
              <input id="input-2" matInput placeholder="placeholder">
            </mat-form-field>
          </div>
          <div class="row-15 no-leading-track">
            <p>
              <a href="http://www.cupcakeipsum.com">Cupcake ipsum:</a>
              Ice cream liquorice halvah danish sesame snaps liquorice. Dragée jelly chocolate chocolate cake.
              Candy canes brownie candy halvah. Chocolate cake marzipan croissant tiramisu tiramisu cheesecake powder
              soufflé sugar plum. Donut tart dragée soufflé croissant. Chupa chups cheesecake marshmallow cheesecake
              candy soufflé cookie. Soufflé marshmallow candy chocolate ice cream tiramisu cookie. Chocolate bar caramels
              jelly beans. Biscuit tootsie roll gingerbread gummies carrot cake sweet roll powder lollipop. Macaroon
              pie donut oat cake wafer jujubes lemon drops fruitcake lollipop. Dragée macaroon brownie gummi bears
              jelly caramels marzipan carrot cake. Jelly jelly beans lollipop bear claw soufflé.
            </p>
          </div>
        </div>
      </sg-section>
    </div>`,
  styles: [
    `
    .sg-container {
      border: 1px solid black;
    }
  `
  ]
});

basic.argTypes = { headerText };
