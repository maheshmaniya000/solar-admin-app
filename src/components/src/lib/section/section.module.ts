import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { SectionComponent } from './section.component';

@NgModule({
  declarations: [SectionComponent],
  imports: [CommonModule],
  exports: [SectionComponent]
})
export class SectionModule {}
