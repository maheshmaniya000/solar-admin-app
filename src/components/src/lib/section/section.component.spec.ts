import { createHostFactory, SpectatorHost } from '@ngneat/spectator/jest';

import { SectionComponent } from './section.component';

describe('SectionComponent', () => {
  describe('component', () => {
    const createHost = createHostFactory({
      component: SectionComponent
    });
    let spectator: SpectatorHost<SectionComponent>;

    beforeEach(() => {
      spectator = createHost(`
        <sg-section>
          <section>Some transcluded <b>content</b></section>
        </sg-section>`);
    });

    it('should match snapshot with transcluded section without header', () => {
      expect(spectator.fixture).toMatchSnapshot();
    });

    it('should match snapshot with transcluded section with header', () => {
      spectator.setInput({
        headerText: 'Some header'
      });
      expect(spectator.fixture).toMatchSnapshot();
    });
  });
});
