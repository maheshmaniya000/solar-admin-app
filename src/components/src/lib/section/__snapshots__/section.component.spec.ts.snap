// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionComponent component should match snapshot with transcluded section with header 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <sg-section>
    <h6
      class="header"
    >
      Some header
    </h6>
    <section>
      <section>
        Some transcluded 
        <b>
          content
        </b>
      </section>
    </section>
  </sg-section>
</ng-component>
`;

exports[`SectionComponent component should match snapshot with transcluded section without header 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <sg-section>
    <section>
      <section>
        Some transcluded 
        <b>
          content
        </b>
      </section>
    </section>
  </sg-section>
</ng-component>
`;
